clustering/marshalling/protostream/src/main/java/org/wildfly/clustering/marshalling/protostream/EnumMarshaller.java

```diff
@@ -22,15 +22,16 @@
 
 package org.wildfly.clustering.marshalling.protostream;
 
-import java.util.OptionalInt;
+import java.io.IOException;
 
-import org.infinispan.protostream.ImmutableSerializationContext;
+import org.infinispan.protostream.impl.WireFormat;
 
 /**
  * ProtoStream marshaller for enums.
  * @author Paul Ferraro
  */
-public class EnumMarshaller<E extends Enum<E>> implements org.infinispan.protostream.EnumMarshaller<E>, Predictable<E> {
+public class EnumMarshaller<E extends Enum<E>> implements org.infinispan.protostream.EnumMarshaller<E>, ProtoStreamMarshaller<E> {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names shouldn’t shadow simple name of implemented interface (NM_SAME_SIMPLE_NAME_AS_INTERFACE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-shouldn-t-shadow-simple-name-of-implemented-interface-nm-same-simple-name-as-interface).
This class/interface has a simple name that is identical to that of an implemented/extended interface, except that the interface is in a different package (e.g., alpha.Foo extends beta.Foo). This can be exceptionally confusing, create lots of situations in which you have to look at import statements to resolve references and creates many opportunities to accidentally define methods that do not override methods in their superclasses.